### PR TITLE
Refactor: restructure concurrency primitives 2

### LINKED ISSUE #2868: Restructure concurrency primitives
Move:

- [ ] `Condition-> Concurrent.Condition`
- [ ] `ReentrantLock` -> Concurrent.ReentrantLock`
- [ ] `Ticker` -> Concurrent.Channel.Ticker`
- [ ] `Timer` -> Concurrent.Channel.Timer`
- [ ] `Channel -> Concurrent.Channel`.

This means both files and namespaces.

### PR DESCRIPTION
**Pull request by ninaandrup**
Fixes #2868

### CODE REVIEW

File: main/src/library/Concurrent/Condition.flix [hunk 1]
```diff
@@ -17,18 +17,17 @@
 ///
 /// A wrapper around a Java Condition.
 ///
-opaque type Condition = ##java.util.concurrent.locks.Condition
+pub opaque type Condition = ##java.util.concurrent.locks.Condition
```
Comment (ninaandrup) [inline main/src/library/Concurrent/Condition.flix:20]: I left this outside of the namespace Concurrent/Condition.

File: main/src/library/Concurrent/ReentrantLock.flix [hunk 2]
```diff
@@ -14,41 +14,41 @@
...
-/// A wrapper around a Java ReentrantLock.
-///
-opaque type ReentrantLock = ##java.util.concurrent.locks.ReentrantLock
+namespace Concurrent/ReentrantLock {
 
-namespace ReentrantLock {
+    ///
+    /// A wrapper around a Java ReentrantLock.
+    ///
+    pub opaque type ReentrantLock = ##java.util.concurrent.locks.ReentrantLock
```
Comment (ninaandrup) [inline main/src/library/Concurrent/ReentrantLock.flix:22]: I put this inside the namespace Concurrent/ReentrantLock.

**Review by ninaandrup — COMMENTED** [review]
I think the `opaque type`s in `Condition` and `ReentrantLock` should be either both be placed inside or outside of the namespaces. But I was not sure. What should I do?

**Comment by JonathanStarup** [comment]
> I think the `opaque type`s in `Condition` and `ReentrantLock` should be either both be placed inside or outside of the namespaces. But I was not sure. What should I do?

I think they should both be inside the namespaces yeah 👍

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**